amethyst_assets/src/loader.rs

```diff
@@ -163,10 +163,11 @@ impl Loader {
     fn context<C>(&self) -> &Arc<C>
         where C: Context + 'static,
     {
-        let context = self.contexts
+        let context: &Box<Any + Send + Sync> = self.contexts
             .get(&TypeId::of::<C::Asset>())
             .expect("Assets need to be registered with `Loader::register`.");
 
+        let context: &Any = context; // `Any + Send + Sync` doesn't has `downcast_ref`
```

Comment:
I think the explicit cast shouldn't be necessary, rustc has enough information to figure out the intent.  I filed an issue [here](https://github.com/rust-lang/rust/issues/44064), let's see if the rust devs agree with me on this one or not.